frontend/ui/widget/datetimewidget.lua

```diff
@@ -101,67 +139,180 @@ function DateTimeWidget:init()
+    if self.month then
+        month_widget = NumberPickerWidget:new{
+            show_parent = self,
+            value = self.month,
+            value_min = self.month_min or 1,
+            value_max = self.month_max or 12,
+            value_step = 1,
+            value_hold_step = self.month_hold_step or 3,
+            width = number_picker_widgets_width,
+        }
+        self:mergeLayoutInHorizontal(month_widget)
+        unit_text = unit_text .. ", " .. times[2]
+    else
+        month_widget = dummy_widget
+    end
+    if self.day then
+        day_widget = NumberPickerWidget:new{
+            show_parent = self,
+            value = self.day,
+            value_min = self.day_min or 1,
+            value_max = self.day_max or 31,
+            value_step = 1,
+            value_hold_step = self.day_hold_step or 3,
+            width = number_picker_widgets_width,
+        }
+        self:mergeLayoutInHorizontal(day_widget)
+        unit_text = unit_text .. ", " .. times[3]
+    else
+        day_widget = dummy_widget
+    end
+
+    if self.hour then
+        hour_widget = NumberPickerWidget:new{
+            show_parent = self,
+            value = self.hour,
+            value_min = self.hour_min or 0,
+            value_max = self.hour_max or 23,
+            value_step = 1,
+            value_hold_step = self.hour_hold_step or 4,
+            width = number_picker_widgets_width,
+        }
+        self:mergeLayoutInHorizontal(hour_widget)
+        unit_text = unit_text .. ", " .. times[4]
+    else
+        hour_widget = dummy_widget
+    end
+    if self.min then
+        min_widget = NumberPickerWidget:new{
+            show_parent = self,
+            value = self.min,
+            value_min = self.min_min or 0,
+            value_max = self.min_max or 59,
+            value_step = 1,
+            value_hold_step = self.min_hold_step or 10,
+            width = number_picker_widgets_width,
+        }
+        self:mergeLayoutInHorizontal(min_widget)
+        unit_text = unit_text .. ", " .. times[5]
+    else
+        min_widget = dummy_widget
+    end
+    if self.sec then
+        sec_widget = NumberPickerWidget:new{
+            show_parent = self,
+            value = self.sec,
+            value_min = self.sec_min or 0,
+            value_max = self.sec_max or 59,
+            value_step = 1,
+            value_hold_step = self.sec_hold_step or 10,
+            width = number_picker_widgets_width,
+        }
+        self:mergeLayoutInHorizontal(sec_widget)
+        unit_text = unit_text .. ", " .. times[6]
+    else
+        sec_widget = dummy_widget
+    end
+
+    -- remove first comma and append a period.
+    unit_text = unit_text:gsub(" ,", "") .. "."
+
+    -- replace last comma with "and"
+    local pos_and = unit_text:find(", %S*$") -- find last comma
+    if pos_and then
+        unit_text = unit_text:sub(1, pos_and-1) .. " " .. _("and") .. unit_text:sub(pos_and+1)
+    end
```

Comment:
Not sure how this will do/allow with translations (feels like western sentence gramar overengineering :)
Do we really need "Time in H, M and S" (which it looks like this build) ?

Reply:
Not sure about translations, but we use DTW for hh:mm and for mm:ss. The  spinners look the same with a colon in between.

Do we want an appended info text like:
"Time is in the format hh:mm." or
"Time is in the format dd/hh:mm."?
Would that make any difference for translation?

Or would it be more consistant to drop the separator and add the unit to the numberpicker value directly ?
So that we get "3 h" "5 m" in the pickers text fields? But what shall we do then with mixed 1d  4 h  2m. And 3 mo 4 d 6 m is weird, and translation even more uncertain. So this is no solution.

Reply:
No strong feeling.
May be `T(_("Time is in the format: %1", fmt)` with `fmt = "hh:mm:ss"` (not translated) could be just fine ?

Reply:
`hh:mm:ss` may be ISO, but surely Dutch is hardly unique in using something like `uu:mm:ss` in more regular contexts.

But in any case, surely better to leave hh:mm:ss than this weird clever code construction with and. Never be clever. ;-)

Reply:
As we dont have so many of them, why not drop the append_unit_info  functionality at all and write proper info_texts?